File: 1958_checkIfMoveIsLegal/main.cpp
#include <vector>

class Solution {
public:
    bool checkMove(std::vector<std::vector<char>>& board, int rMove, int cMove, char color) {
        int ROWS=board.size();
        int COLS=board[0].size();
        
        std::vector<std::vector<int>> dirs={{0,1},{1,0},{-1,0},{0,-1},{1,1},{1,-1},{-1,1},{-1,-1}}; 
        
        char opposition = color=='W' ? 'B' : 'W'; //opposite color 
        
        for(int i = 0; i < 8; ++i) {
            int row = rMove + dirs[i][0];
            int col = cMove + dirs[i][1];
            int count = 1;
            
            while(row < ROWS && col < COLS && row >= 0 && col >= 0 && board[row][col] != '.'){
                count++;
                if(count >= 3 && board[row][col] == color) 
                    return true;
                
                if(board[row][col] != opposition) 
                    break;
                
                row += dirs[i][0];
                col += dirs[i][1];
            }
        }
        return false;
    }
};

File: 1958_checkIfMoveIsLegal/main.py
from typing import List
class Solution:
    def checkMove(self, board: List[List[str]], rMove: int, cMove: int, color: str) -> bool:
        ROWS, COLS = len(board), len(board[0])
        dir = [[1, 0], [-1, 0], [0, 1], [0, -1],
               [1, 1], [-1, -1], [1, -1], [-1, 1]]
        board[rMove][cMove] = color

        def legal(row, col, color, direction):
            row_dir, col_dir = direction
            row, col = row + row_dir, col + col_dir
            length = 1

            while( 0 <= row < ROWS and  0 <= col < COLS):
                length += 1
                if board[row][col] == ".":
                    return False
                if board[row][col] == color:
                    return length >= 3
                row, col = row + row_dir, col + col_dir
            return False
        
        for d in dir:
            if legal(rMove, cMove, color, d):
                return True
        return False